### PR TITLE
[NCJS-9] Features/option network byte

### PR DESCRIPTION
**Pull request by j0k**
networkByte + README 

networkByte still doesn't checked in;
```
export class Transaction {
  static fromJSON(obj: any): Transaction {
```

### CODE REVIEW

File: src/transaction.ts [hunk 1]
```diff
@@ -87,21 +87,31 @@ export class Transaction {
       assertAddressValid(o.address);
       assertAmountValid(o.amount);
     });
-    assertByteValid(networkByte);
+
+    if (networkByte !== undefined) assertByteValid(networkByte);
```
Comment (Warchant) [inline src/transaction.ts]:
```suggestion
    if (!networkByte) assertByteValid(networkByte);
```

File: index.html [hunk 2]
```diff
@@ -309,14 +309,17 @@ <h2 style="text-align:center">test page for nodecore-js <a href="https://github.
...
+          var signature = VeriBlock.crypto.SHA256withECDSA.sign(msg, pk)
+          // var message = data.message;
+          // var pk = VeriBlock.crypto.PrivateKey.fromStringBase64(data.privKey64);
+          // var tx = VeriBlock.transaction.Transaction.parse(data.transaction);
+          // var kp = VeriBlock.crypto.KeyPair.fromPrivateKey(pk);
+          // var signature = VeriBlock.transaction.signTransaction(tx, kp, data.signatureIndex);
+          data.signature = signature.toStringHex(); //signature.asn1.toString('base64');
         } catch(e) {
-          alert(`Invalid tansaction: ${JSON.stringify(e)}`);
+          alert(`Invalid signing: ${JSON.stringify(e)}`);
```
Comment (Warchant) [inline index.html]:
```suggestion
          alert(`Invalid signature: ${JSON.stringify(e)}`);
```

File: src/transaction.ts [hunk 3]
```diff
@@ -87,21 +87,31 @@ export class Transaction {
...
 
   toJSON() {
-    return {
-      type: this.type,
-      sourceAddress: this.sourceAddress,
-      sourceAmount: makeBigNumber(this.sourceAmount),
-      outputs: this.outputs,
-      networkByte: this.networkByte,
-    };
+    if (this.networkByte === undefined) {
```
Comment (Warchant) [inline src/transaction.ts]: change to 
```
const obj = {
...
// no networkByte
}

if(this.networkByte){
  obj["networkByte"] = this.networkByte;
}

return obj;
```

File: src/transaction.ts (hunk 1 above)
Comment (j0k) [reply src/transaction.ts]: `if (!networkByte) assertByteValid(networkByte);`
cause 
```
src/transaction.ts:99:39 - error TS2345: Argument of type 'number | undefined' is not assignable to parameter of type 'number'.
  Type 'undefined' is not assignable to type 'number'.

99     if (!networkByte) assertByteValid(networkByte);
                                         ~~~~~~~~~~~
```

**Pull request merged**